graph/src/data_source/mod.rs

```diff
@@ -42,6 +42,33 @@ pub enum DataSourceCreationError {
     Unknown(#[from] Error),
 }
 
+/// Which entity types a data source can read and write to.
+pub enum EntityTypeAccess {
```

Comment:
Might be worth adding a bit more here about how this interacts with causality region based isolation

Reply:
I've written a more detailed comment.